### PR TITLE
chakrashim: fixing ObjectTemplate implementation

### PR DESCRIPTION
**Pull request by kfarnung**
It's possible to create an object template with a constructor in V8,
but we didn't support it.  It's a little weird (even in V8) since
objects constructed this way will never trigger the CallHandler of the
FunctionTemplate object, it just inherits the prototype from it.

##### Checklist

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/CONTRIBUTING.md#commit-message-guidelines)

##### Affected core subsystem(s)

chakrashim

### CODE REVIEW

**Comment by obastemur** [comment]
@kfarnung this was broken by design? What we gain by implementing? ie perf/memory effect? How to test it?

**Comment by kfarnung** [comment]
I don't think this was broken by design, it looks like this is just an edge case implementation detail that wasn't implemented previously. This behavior is required in the latest http2 changes from upstream. I tried to minimize the impact and have a local test suite that I'm using to validate the behavior against V8.

File: deps/chakrashim/src/v8objecttemplate.cc [hunk 1]
```diff
@@ -868,6 +869,25 @@ Local<Object> ObjectTemplate::NewInstance(Handle<Object> prototype) {
...
+                          iso->GetCachedPropertyIdRef(
+                              jsrt::CachedPropertyIdRef::prototype),
+                          &prototypeValue) == JsNoError) {
+          prototype = Local<Object>::New(prototypeValue);
+        }
+      }
+    }
+  }
+
   if (!prototype.IsEmpty()) {
```
Comment (MSLaguana) [inline deps/chakrashim/src/v8objecttemplate.cc:893]: Is there a reason you chose not to make this an `else`?

I'd also like to see a comment here explaining the scenario this is supporting; it's something that I could easily see being broken if it is forgotten.

Comment (kfarnung) [reply deps/chakrashim/src/v8objecttemplate.cc:893]: The reason is that the previous conditional is just an alternate way to get the prototype.  It could have already been provided directly.

Comment (MSLaguana) [reply deps/chakrashim/src/v8objecttemplate.cc:893]: Ah, I'd missed that it was setting `prototype`

**Review by MSLaguana — APPROVED** [review]

**Comment by kfarnung** [comment]
CI: https://ci.nodejs.org/job/chakracore-test-pull-request/165/
